V4.0.0 - Move metrics hash out of script template, add orch-services, and add PuppetDB metrics

npwalker opened the pull request:
This PR includes https://github.com/npwalker/pe_metric_curl_cron_jobs/pull/20.  We can either merge this one or rebase it after https://github.com/npwalker/pe_metric_curl_cron_jobs/pull/20 is merged.

npwalker committed:
Add collection of orchestration-services status endpoint

This commit adds metric collection for orchestration-services.
The status endpoint on orchestration-services gather pcp-broker
information as well.

npwalker committed:
Refactor puppetserver metrics into their own class

This commit moves puppetserver metrics into their own class so that
we can make it easier to add new parameters and customizations
per metric type in the future without adding every new parameter
to init.pp.

npwalker committed:
Refactor puppetdb metrics into their own class

This commit moves puppetdb metrics into their own class so that
we can make it easier to add new parameters and customizations
per metric type in the future without adding every new parameter
to init.pp.

npwalker committed:
Refactor activemq metrics into their own class

This commit moves activemq metrics into their own class so that
we can make it easier to add new parameters and customizations
per metric type in the future without adding every new parameter
to init.pp.

npwalker committed:
Refactor orchestrator metrics into their own class

This commit moves orchestrator metrics into their own class so that
we can make it easier to add new parameters and customizations
per metric type in the future without adding every new parameter
to init.pp.

npwalker committed:
Move metrics hash out of the template and into puppet code

Prior to this commit, we had a metrics hash for puppetdb in the
tk_metrics template.  Since puppetserver and orchestrator use that
template it wasn't the right place for the metrics hash.

After this commit, the metrics hash is passed in via a parameter
to the template and PuppetDB figures out the hash to pass in via
its own metrics class.

npwalker committed:
Produce an actual array of hashes in the script

Prior to this commit, trying to just output the puppet array
of hashes removes all the qouting that is needed.

npwalker committed:
Only collect puppetdb activemq metrics in 2015.y versions of PE

Apparently the rest of the metrics had different names in 2015 so
they just result in a bunch of 'could not find mbean' errors.

jarretlavallee reviewed (changes requested):
It looks good. Adding the additional metrics will be beneficial for future debugging. A couple things that I found are below.

1. ~~*additional_metrics* default of an empty array in *pe_metric.pp* throws a lambda error in 2017.1.1~~
2. Can we stagger the tidy cron jobs by a few minutes to reduce the load on the master at compression time.

npwalker committed:
Add HakariCP connection pool metrics

We added HakariCP in PE2016.1 and it has metrics that tell you
the number of active, pending, and idle connections which would
be good to know if you are out of available connections.

npwalker committed:
Add all global metrics to puppetdb metrics

Also, rename command_processing_time to global_processing_time and command_processed to global_processed which aligns with the rest of
the names and the

npwalker committed:
Add puppetdb storage metrics

This commit adds all of the puppetdb storage metrics except for
those that reach out to the database to execute queries.

npwalker committed:
Rename the mq metrics to include the namespace

This commit renames the mq metrics so it is more clear where they
came from.

npwalker committed:
Add 4.0.0 changelog and bump version in metadata.json

jarretlavallee commented:
:+1:
